### PR TITLE
unit-tests-jest

### PR DESCRIPTION
**Pull request by TristanChao**

### CODE REVIEW

File: unit-tests-jest/numbers.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { evenNumbers, toDollars, divideBy, multiplyBy } from './numbers';
+
+test('filters array to only contain even numbers', () => {
+  expect(evenNumbers([4.4, 8, 19, 7])).toStrictEqual([8]);
+});
+
+test('converts a number into $#.## format', () => {
+  expect(toDollars(5.3333333)).toBe('$5.33');
+});
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:11]: Any edge cases you can think of?

File: unit-tests-jest/numbers.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+test('divides each number in an array (first arg) by the divisor (second arg)', () => {
+  expect(divideBy([4, 5, 6, 7, 8], 2)).toStrictEqual([2, 2.5, 3, 3.5, 4]);
+});
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:22]: Any edge cases? How about checking if the array has been modified?

File: unit-tests-jest/numbers.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Returns a new array of numbers where every entry has been divided by the
+ * given divisor. Does not modify the original array.
+ */
+export function divideBy(numbers: number[], divisor: number): number[] {
+  for (let i = 0; i < numbers.length; i++) {
+    numbers[i] = numbers[i] / divisor;
+  }
+  return numbers;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: This modifies the array, but the description says it should not modify it.

File: unit-tests-jest/numbers.test.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+test('in an object (1st arg) all numerical values are multiplied by multiplier (2nd arg)', () => {
+  const test = { first: 2, second: 'eight', third: 9 };
+  const expected = { first: 6, second: 'eight', third: 27 };
+  expect(multiplyBy(test, 3)).toStrictEqual(expected);
+});
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:42]: Any edge cases? How about non-integers?

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: unit-tests-jest/numbers.ts [hunk 5]
```diff
@@ -9,18 +9,23 @@ export function evenNumbers(numbers: number[]): number[] {
...
 /**
  * Returns a new array of numbers where every entry has been divided by the
  * given divisor. Does not modify the original array.
  */
-export function divideBy(numbers: number[], divisor: number): number[] {
-  for (let i = 0; i < numbers.length; i++) {
-    numbers[i] = numbers[i] / divisor;
-  }
-  return numbers;
+export function divideBy(numbers: number[], divisor: number): number[][] {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: Why did you change the return type? There's no need to return the original array, you could just return `numbers.map`.

File: unit-tests-jest/numbers.ts [hunk 6]
```diff
@@ -33,8 +38,8 @@ export function multiplyBy(
 ): Record<string, unknown> {
   const result: Record<string, unknown> = {};
   Object.entries(obj).forEach(([key, value]) => {
-    if (Number.isInteger(value)) {
-      result[key] = Number(value) * multiplier;
+    if (!Number.isNaN(Number(value))) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: The easiest check is `typeof value !== 'number'`.

File: unit-tests-jest/numbers.ts [hunk 7]
```diff
@@ -33,8 +38,8 @@ export function multiplyBy(
...
+      result[key] = Math.round(Number(value) * multiplier * 100) / 100;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts]: I don't understand this math? Why not just `Number(value) * multiplier`? (Or, since you know it's a number, just `value * multiplier`.)

**Review by RobertGardner — APPROVED** [review]
Minor comments. You have the unit tests correct, and that's the important part.
